src/modalities/optimizers/optimizer_factory.py

```diff
@@ -39,13 +39,16 @@ def get_adam_w(
         return optimizer
 
     @staticmethod
-    def get_checkpointed_optimizer(
-        checkpoint_loading: CheckpointLoadingIF, checkpoint_path: Path, wrapped_model: nn.Module, optimizer: Optimizer
+    def get_checkpointed_optimizer_(
+        checkpoint_loading: FSDP1CheckpointLoadingIF,
+        checkpoint_path: Path,
+        wrapped_model: nn.Module,
+        optimizer: Optimizer,
     ) -> Optimizer:
-        wrapped_optimizer = checkpoint_loading.load_optimizer_checkpoint(
+        checkpoint_loading.load_optimizer_checkpoint_(
```

Comment:
Is the checkpoint loaded in-place?

Reply:
We need to return this, since it is used as a component. 
https://github.com/Modalities/modalities/blob/ef2f8c27ae6b36c65e5cf3600127e3cc3f0b033e/src/modalities/registry/components.py#L172